### ISSUE TITLE
`C416` doesn't catch dictionary comprehensions with tuple targets

### ISSUE THREAD

**Issue by adamtheturtle** [astral-sh/ruff#7307]
`pylint 2.17.5` shows a R1721 error with the following file:

```python
# example.py
"""Code which has R1721 error in pylint but not ruff."""

[(k, v) for k, v in {}.items()]

```

ruff 0.0.288 shows no error for this:

```
ruff --select=ALL --isolated example.py
```

I expect `ruff` to show an error in this case.
https://github.com/astral-sh/ruff/issues/970 has `unnecessary-comprehension / R1721` checked as done.

**Comment by charliermarsh** [comment]
`R1721` is implemented as a different rule: https://beta.ruff.rs/docs/rules/unnecessary-comprehension/ (C416). I just updated the issue to reflect that. However, `C416` isn't catching this case. I think it should.
